### ISSUE TITLE
error: 'switch' may only be used as expression in return, throw, or as the source of an assignment

### ISSUE THREAD

**Issue by everlof** [nicklockwood/SwiftFormat#1548]
```
$ swiftformat --version
0.52.6
```

This code: 
```
private func constraint() -> [Int] {
    [
        1,
        2,
        {
            switch Bool.random() {
            case true: 3
            case false: 4
            }
        }(),
    ]
}
```

generates this code: 

```
private func constraint() -> [Int] {
    [
        1,
        2,
        switch Bool.random() {
        case true: 3
        case false: 4
        },
    ]
}
```

which doesn't compile due to:

```
file.swift:5:9: error: 'switch' may only be used as expression in return, throw, or as the source of an assignment
        switch Bool.random() {
        ^
```

**Comment by nicklockwood** [comment]
@calda would you mind taking a look at this one?

**Comment by nicklockwood** [comment]
@everlof fixed in 0.52.8

**Comment by everlof** [comment]
Awesome! Thanks!
